apps/lockfile-explorer-web/src/containers/PackageJsonViewer/index.tsx

```diff
@@ -9,47 +15,133 @@ enum PackageView {
 }
 
 export const PackageJsonViewer = (): JSX.Element => {
-  const [packageJSON, setPackageJSON] = useState('');
-  const [parsedPackageJSON, setParsedPackageJSON] = useState('');
+  const [packageJSON, setPackageJSON] = useState<IPackageJson | undefined>(undefined);
+  const [parsedPackageJSON, setParsedPackageJSON] = useState<IPackageJson | undefined>(undefined);
+  const [specChanges, setSpecChanges] = useState<Map<string, ISpecChange>>(new Map());
   const [cjs, setCjs] = useState('');
+  const selectedEntry = useAppSelector(selectCurrentEntry);
 
-  const [selection, setSelection] = useState<PackageView>(PackageView.PACKAGE_JSON);
+  const [selection, setSelection] = useState<PackageView>(PackageView.PARSED_PACKAGE_JSON);
 
   const cb = useCallback((s: PackageView) => () => setSelection(s), []);
 
   useEffect(() => {
-    async function loadPackageDetails(): Promise<void> {
-      const cjsFile = await readCJS();
+    async function loadPackageDetails(packageName: string): Promise<void> {
+      const cjsFile = await readPnpmfile();
       setCjs(cjsFile);
-      const packageJSONFile = await readPackageJSON();
-      setPackageJSON(packageJSONFile as string);
-      const parsedJSON = await readParsedCJS();
-      setParsedPackageJSON(parsedJSON as string);
+      const packageJSONFile = await readPackageJSON(packageName);
+      setPackageJSON(packageJSONFile);
+      const parsedJSON = await readParsedCJS(packageName);
+      setParsedPackageJSON(parsedJSON);
+
+      if (packageJSONFile && parsedJSON) {
+        const diffDeps = compareSpec(packageJSONFile, parsedJSON);
+        setSpecChanges(diffDeps);
+      }
+    }
+    if (selectedEntry) {
+      if (selectedEntry.entryPackageName) {
+        /* eslint @typescript-eslint/no-floating-promises: off */
+        loadPackageDetails(selectedEntry.packageJsonFolderPath);
+      } else {
+        console.log('selected entry has no entry name: ', selectedEntry.entryPackageName);
```

Comment:
```suggestion
        console.log('The selected entry has no entry name: ', selectedEntry.entryPackageName);
```

Just curious, how might this arise?  Might be useful to add a code comment with some context

Reply:
This was mainly used to detect mistakes in our parsing logic and theoretically shouldn't happen with a valid lockfile

Reply:
Added a comment to explain